services/streammanager/kafka/kafkamanager.go

```diff
@@ -234,6 +235,19 @@ func Init() {
 		)
 	}
 
+	kafkaCompression = client.CompressionZstd
```

Comment:
Should we enable compression by default only if batching is enabled too?

Reply:
Yeah I thought about it but I wasn't going to bother because I wanted to propose to enable batching for everybody given that it doesn't really affect the customers. They don't need to do anything special. We batch at the processor, then the transformer takes care of it and we just end up with an array in the kafka manager so I don't see why we shouldn't just enable compression plus batching for everybody.

Nonetheless the setting would be possible to use so it's a sensible choice not to compress if batching is disabled. Will do the change :+1: